pkg/migrate/migrate.go

```diff
@@ -1146,3 +1296,249 @@ func resetReclaimPolicy(ctx context.Context, w *log.Logger, clientset k8sclient.
 
 	return nil
 }
+
+// buildPVConsumerPod creates a pod spec for consuming a pvc
+func (p *PVMigrator) buildPVConsumerPod(pvcName string) *corev1.Pod {
+	podName := p.podNameOverride
+	if podName == "" {
+		tmp := uuid.New().String()[:5]
+		podName = fmt.Sprintf("pvmigrate-vol-consumer-%s-%s", pvcName, tmp)
+	}
+	if len(podName) > 63 {
+		podName = podName[0:31] + podName[len(podName)-32:]
+	}
+	return &corev1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      podName,
+			Namespace: "default",
+		},
+		Spec: corev1.PodSpec{
+			RestartPolicy: corev1.RestartPolicyNever,
+			Volumes: []corev1.Volume{
+				{
+					Name: "tmp",
+					VolumeSource: corev1.VolumeSource{
+						PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+							ClaimName: pvcName,
+						},
+					},
+				},
+			},
+			Containers: []corev1.Container{
+				{
+					Name:  "busybox",
+					Image: "busybox",
+					Command: []string{
+						"sleep",
+						"3600",
+					},
+					VolumeMounts: []corev1.VolumeMount{
+						{
+							MountPath: "/tmpmount",
+							Name:      "tmp",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+// buildPVC creates a temporary PVC requesting for 1Mi of storage for a provided storage class name.
+func (p *PVMigrator) buildTmpPVC(pvc corev1.PersistentVolumeClaim, sc string) *corev1.PersistentVolumeClaim {
+	pvcName := p.pvcNameOverride
+	if pvcName == "" {
+		tmp := uuid.New().String()[:5]
+		pvcName = fmt.Sprintf("pvmigrate-claim-%s-%s", pvc.Name, tmp)
+	}
+	if len(pvcName) > 63 {
+		pvcName = pvcName[0:31] + pvcName[len(pvcName)-32:]
+	}
+
+	return &corev1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      pvcName,
+			Namespace: "default",
+		},
+		Spec: corev1.PersistentVolumeClaimSpec{
+			StorageClassName: &sc,
+			AccessModes:      pvc.Spec.AccessModes,
+			Resources: corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					corev1.ResourceStorage: resource.MustParse("1Mi"),
+				},
+			},
+		},
+		// for testing/mocking
+		// This fields gets set and updated by the kube api server
+		Status: corev1.PersistentVolumeClaimStatus{Phase: pvc.Status.Phase},
```

Comment:
If you're doing this so that you can set the claim status for something else to read in a test function, I'd recommend looking at the `backgroundfunc` stuff in some of the other test functions - that way you can wait for 1/10th of a second or whatever and ensure that the logic does not require status to be immediately perfect